### ISSUE TITLE
Sanity check the newly created Release 1.5.0 RPC endpoints and their responses

### ISSUE THREAD

**Issue by devendran-m** [casper-ecosystem/casper-js-sdk#161]
Sanity check the newly created Release 1.5.0 RPC endpoints and their responses

**Comment by hoffmannjan** [comment]
~~`chain_get_state_root_hash` stoped working for~~
```
params: {
  block_hash: blockHashBase16 || null
}
```
~~For this params it returns `Failed to parse 'params' field: invalid type: null, expected string or map`.~~

It's untrue as this is a wider problem described below.

**Comment by hoffmannjan** [comment]
I created an [issue](https://github.com/open-rpc/client-js/issues/314) in `@open-rpc/client-js` repo, but nobody answered yet. 

I also did a check, and [this library](https://www.npmjs.com/package/json-rpc-2.0) `json-rpc-2.0` works as we want, seems to be reasonable (still pretty popular, 8,525 downloads / week) but it has a little bit different API. 

That means, I will need to rework that part of SDK to swap the library and test it properly. Also, from what I know some users might depend on Provider functionality (at least [Tor.us](http://tor.us/) is), which here is named Middleware and it’s a bit different, so we will need to be very careful communicating the changes to the users.

So my guess is we need to the decide which way is better:

* make the 1.5 compatible with previous SDK and treat params: [] same way as params: null or missing params
* swap the library in js-sdk, would be also good to check other SDKs as this is something that potentially can break them

I should be able to do the swap in a few days including testing, so on Friday we should have working release-candidate that can be shared with MAKE and [Tor.us](http://tor.us/) at least. (edited)
